src/swarm-util/Swarm/Util/OccurrenceEncoder.hs

```diff
@@ -0,0 +1,47 @@
+-- |
+-- SPDX-License-Identifier: BSD-3-Clause
+-- Description: Compress representation of traversable
+--
+-- For structures with many repeating elements,
+-- will store the expanded element representation
+-- in an array so that the structure can be transmitted
+-- with array indices as elements.
```

Comment:
This description and comment did not make much sense to me.  What is a "traversable" in this context?  What is a "structure"?  What is an "expanded element representation"?  It seems to me that this module is doing something fairly straightforward: keep track of seen items with a unique index for each, allowing us to quickly look up the index of an item or assign a new index to an item we have not seen before.  I'm guessing that you were writing these descriptions with the *intended use case* in mind, but I think prior to that we should just write a simple description of what it actually does (and then optionally include a description of how it is used).

Reply:
Yes, the struggle with module documentation was a sign that the API was not good.  I've simplified it a bit.